pkg/azure/access/helpers/vm.go

```diff
@@ -0,0 +1,224 @@
+	getResp, err = vmClient.Get(ctx, resourceGroup, vmName, nil)
+	if err != nil {
+		if errors.IsNotFoundAzAPIError(err) {
+			return nil, nil
+		}
+		return
+	}
+	vm = &getResp.VirtualMachine
+	return
+}
+
+// DeleteVirtualMachine deletes the Virtual Machine with the give name and belonging to the passed in resource group.
+// If cascade delete is set for associated NICs and Disks then these resources will also be deleted along with the VM.
+// NOTE: All calls to this Azure API are instrumented as prometheus metric.
+func DeleteVirtualMachine(ctx context.Context, vmAccess *armcompute.VirtualMachinesClient, resourceGroup, vmName string) (err error) {
+	defer instrument.RecordAzAPIMetric(err, vmDeleteServiceLabel, time.Now())
+	delCtx, cancelFn := context.WithTimeout(ctx, defaultDeleteVMTimeout)
+	defer cancelFn()
+	poller, err := vmAccess.BeginDelete(delCtx, resourceGroup, vmName, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Failed to trigger delete of VM [ResourceGroup: %s, VMName: %s]", resourceGroup, vmName)
+		return
+	}
+	_, err = poller.PollUntilDone(delCtx, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Polling failed while waiting for delete of VM: %s for ResourceGroup: %s", vmName, resourceGroup)
+		return
+	}
+	return
+}
+
+// CreateVirtualMachine creates a Virtual Machine given a resourceGroup and virtual machine creation parameters.
+// NOTE: All calls to this Azure API are instrumented as prometheus metric.
+func CreateVirtualMachine(ctx context.Context, vmAccess *armcompute.VirtualMachinesClient, resourceGroup string, vmCreationParams armcompute.VirtualMachine) (vm *armcompute.VirtualMachine, err error) {
+	defer instrument.RecordAzAPIMetric(err, vmCreateServiceLabel, time.Now())
+	createCtx, cancelFn := context.WithTimeout(ctx, defaultCreateVMTimeout)
+	defer cancelFn()
+	vmName := *vmCreationParams.Name
+	poller, err := vmAccess.BeginCreateOrUpdate(createCtx, resourceGroup, vmName, vmCreationParams, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Failed to trigger create of VM [ResourceGroup: %s, VMName: %s]", resourceGroup, vmName)
+		return
+	}
+	createResp, err := poller.PollUntilDone(createCtx, nil)
+	if err != nil {
+		errors.LogAzAPIError(err, "Polling failed while waiting for create of VM: %s for ResourceGroup: %s", vmName, resourceGroup)
+		return
+	}
+	vm = &createResp.VirtualMachine
+	return
+}
+
+// SetCascadeDeleteForNICsAndDisks sets cascade deletion for NICs and Disks (OSDisk and DataDisks) associated to passed virtual machine.
+// NOTE: All calls to this Azure API are instrumented as prometheus metric.
+func SetCascadeDeleteForNICsAndDisks(ctx context.Context, vmClient *armcompute.VirtualMachinesClient, resourceGroup string, vm *armcompute.VirtualMachine) (err error) {
+	defer instrument.RecordAzAPIMetric(err, vmUpdateServiceLabel, time.Now())
+	vmUpdateParams := createVMUpdateParamsForCascadeDeleteOptions(vm) // TODO: Rename this method it returns a VM not "params"
```

Comment:
lets rename to `createVirtualMachineUpdateDescription`

Reply:
Done, PTAL